cpp/pipelines/collect_journal_stats.cc

```diff
@@ -0,0 +1,213 @@
+    for (const auto &journal : zeder) {
+        if (journal.empty())
+            continue;
+        ++included_journal_count;
+
+        const auto print_ppns(SplitZederPPNs(journal.lookup("pppn")));
+        const auto online_ppns(SplitZederPPNs(journal.lookup("eppn")));
+
+        if (print_ppns.empty() and online_ppns.empty()) {
+            --included_journal_count;
+            LOG_WARNING("Zeder entry #" + std::to_string(journal.getId()) + " is missing print and online PPN's!");
+            continue;
+        }
+
+        for (const auto &print_ppn : print_ppns)
+            ppns_to_zeder_ids_and_types_map.emplace(print_ppn, ZederIdAndType(journal.getId(), 'p'));
+
+        for (const auto &online_ppn : online_ppns)
+            ppns_to_zeder_ids_and_types_map.emplace(online_ppn, ZederIdAndType(journal.getId(), 'e'));
+    }
+
+    LOG_INFO("downloaded information for " + std::to_string(included_journal_count) + " journal(s) from Zeder.");
+
+    return ppns_to_zeder_ids_and_types_map;
+}
+
+
+bool SplitPageNumbers(const std::string &possibly_combined_pages, unsigned * const start_page, unsigned * const end_page) {
+    if (StringUtil::ToUnsigned(possibly_combined_pages, start_page)) {
+        *end_page = *start_page;
+        return true;
+    } else if (possibly_combined_pages.length() >= 2
+               and possibly_combined_pages[possibly_combined_pages.length() - 1] == 'f')
+    {
+        if (not StringUtil::ToUnsigned(possibly_combined_pages.substr(0, possibly_combined_pages.length() - 1), start_page))
+            return false;
+        *end_page = *start_page + 1;
+        return true;
+    }
+
+    // Common case, page range with a hypen.
+    const auto first_hyphen_pos(possibly_combined_pages.find('-'));
+    if (first_hyphen_pos == std::string::npos)
+        return false;
+
+    return StringUtil::ToUnsigned(possibly_combined_pages.substr(0, first_hyphen_pos), start_page)
+           and StringUtil::ToUnsigned(possibly_combined_pages.substr(first_hyphen_pos + 1), end_page);
+}
+
+
+// \return the year as a small integer or 0 if we could not parse it.
+unsigned short YearStringToShort(const std::string &year_as_string) {
+    unsigned short year_as_unsigned_short;
+    if (not StringUtil::ToUnsignedShort(year_as_string, &year_as_unsigned_short))
+        return 0;
+    return year_as_unsigned_short;
+}
+
+void ProcessRecords(MARC::Reader * const reader, const std::string &system_type,
+                    const std::unordered_map<std::string, ZederIdAndType> &ppns_to_zeder_ids_and_types_map,
+                    DbConnection * const db_connection)
+{
+    const auto zeder_instance(system_type == "krimdok" ? "krim" : system_type);
+    const auto JOB_START_TIME(std::to_string(std::time(nullptr)));
+    const auto HOSTNAME(DnsUtil::GetHostname());
+    const std::string ZEDER_URL_PREFIX("http://www-ub.ub.uni-tuebingen.de/zeder/?instanz=" + zeder_instance + "#suche=Z%3D");
```

Comment:
URL is hardcoded here, Zeder::GetFullDumpEndpointPath should be used instead, as well as Zeder::Flavour